server/src/main/java/com/google/coffeehouse/servlets/CreatePersonServlet.java

```diff
@@ -56,40 +62,32 @@ public CreatePersonServlet() {
 
   /** 
    * Creates a {@link Person} object, saves it in the database and returns it in JSON format.
-   * @param request the post request that must have {@code email} and {@code nickname} defined
-   *     as parameters or the respone will send a "400 Bad Request error"
+   * @param request the POST request that must have a valid JSON representation of the Person to be
+   *     created as its body. If this is not the case the response will send a 
+   *     "400 Bad Request error"
    * @param response the response from this method, will contain the created object in JSON format.
-   *     If the request object does not have {@code email} and {@code nickname} parameters, this 
-   *     object will send a "400 Bad Request error"
+   *     If the request object does not have a valid JSON body that describes the Club to be 
+   *     created, this object will send a "400 Bad Request error"
    * @throws IOException if an input or output error is detected when the servlet handles the request
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String email = request.getParameter("email");
-    String nickname = request.getParameter("nickname");
-    String pronouns = request.getParameter("pronouns");
+    String requestBody = request.getReader().lines().collect(Collectors.joining());
 
-    if (email == null || nickname == null) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, EMAIL_OR_NICKNAME_ERROR);
+    Gson gson = new Gson();
```

Comment:
You can probably go a step further and just have one instance of Gson as a class property, that gets created only when CreatePersonServlet is constructed

Reply:
That makes sense, change has been pushed :)